crypto/transactions/builder/base.py

```diff
@@ -38,8 +51,51 @@ def second_sign(self, passphrase):
         self.transaction.signSignature = message.signature
         self.transaction.id = self.transaction.get_id()
 
-    def verify(self):
-        self.transaction.verify()
+    def multi_sign(self, passphrase, index):
+        if not self.transaction.signatures:
+            self.transaction.signatures = []
+        self.set_version(2)
+
+        index = len(self.transaction.signatures) if index == -1 else index
+
+        msg = hashlib.sha256(self.transaction.to_bytes()).digest()
+        secret = unhexlify(PrivateKey.from_passphrase(passphrase).to_hex())
+        signature = hexlify(schnorr.bcrypto410_sign(msg, secret))
+
+        index_formatted = hex(index).replace('x', '')
+        self.transaction.signatures.append(index_formatted + signature.decode())
+
+    def schnorr_verify(self):
+        return self.transaction.verify_schnorr()
+
+    def schnorr_verify_multisig(self):
+        return self.transaction.verify_schnorr_multisig()
+
+    def set_version(self, version=2):
```

Comment:
imo no need to be able to set a version, as v1 is no longer supported with these changes so everything is v2